src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html

```diff
@@ -0,0 +1,190 @@
+                       type="password"
+                       placeholder="Confirm password..."
+                       id="confirmpassword"
+                       name="confirmpassword"
+                       formControlName="confirmpassword">
+                <span class="input-group-btn">
+                  <button type="button"
+                          class="btn btn-default"
+                          cdPasswordButton="confirmpassword">
+                  </button>
+                </span>
+              </div>
+            <span i18n
+                  class="help-block"
+                  *ngIf="(formDir.submitted || userForm.controls.confirmpassword.dirty) && userForm.controls.confirmpassword.hasError('required')">
+              This field is required.
+            </span>
+            <span i18n
+                  class="help-block"
+                  *ngIf="(formDir.submitted || userForm.controls.confirmpassword.dirty) && userForm.controls.confirmpassword.hasError('match')">
+              Password confirmation doesn't match the password.
+            </span>
+          </div>
+        </div>
+
+        <!-- Name -->
+        <div class="form-group"
+             [ngClass]="{'has-error': (formDir.submitted || userForm.controls.name.dirty) && userForm.controls.name.invalid}">
+          <label i18n
+                 class="control-label col-sm-3"
+                 for="name">Full name
+          </label>
+          <div class="col-sm-9">
+            <input class="form-control"
+                   type="text"
+                   placeholder="Full name..."
+                   id="name"
+                   name="name"
+                   formControlName="name">
+          </div>
+        </div>
+
+        <!-- Email -->
+        <div class="form-group"
+             [ngClass]="{'has-error': (formDir.submitted || userForm.controls.email.dirty) && userForm.controls.email.invalid}">
+          <label i18n
+                 class="control-label col-sm-3"
+                 for="email">Email
+          </label>
+          <div class="col-sm-9">
+            <input class="form-control"
+                   type="email"
+                   placeholder="Email..."
+                   id="email"
+                   name="email"
+                   formControlName="email">
+
+            <span i18n
+                  class="help-block"
+                  *ngIf="(formDir.submitted || userForm.controls.email.dirty) && userForm.controls.email.hasError('email')">
+              Invalid email.
+            </span>
+          </div>
+        </div>
+
+        <!-- Roles -->
+        <label class="col-sm-3 control-label"
+               i18n>Roles
+        </label>
+        <div class="col-sm-9">
+          <span class="form-control no-border">
+            <cd-select-badges
+                [data]="userForm.controls.roles.value"
+                [options]="allRoles"
+                emptyMessage="There are no roles."></cd-select-badges>
```

Comment:
I think "There are no roles." can be a bit misleading (no roles available at all).
Maybe it should be rephrased to something like "Currently no roles selected." or "No roles have been selected." ?